### PR TITLE
use segment to track analytics, add time out for paging events, add tracking for modals

### PR DESCRIPTION
**Pull request by phoebemirman**
use segment to track analytics, add time out for paging events, add tracking for modals

add tracking for market creation

moved time out to be in plugin

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://dashboard.featurepeek.com). 📈
Your deployment will be kept up-to-date with this pull request's latest changes.

**https://peek.run/m96h1n5**

-------

*Please read [our docs](https://docs.featurepeek.com/configuring-peek-yml/) for more configuration details.*

**Review by bthaile — CHANGES_REQUESTED** [review]
pull out showFacebookShare and showTwitterShare into their own helper and put the analytics in there.  I'd like to keep the components ignorant of analytics as much as possible. 
Plus I think those methods will be needed in the market cards in the near future.

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://dashboard.featurepeek.com). ⚡️
Your deployment will be kept up-to-date with this pull request's latest changes.

**https://peek.run/zr5hxj7**

-------

*Please read [our docs](https://docs.featurepeek.com/configuring-peek-yml/) for more configuration details.*

File: packages/augur-ui/src/modules/market/components/common/social-media-buttons.tsx [hunk 1]
```diff
@@ -32,17 +35,19 @@ export const SocialMediaButtons = (props: SocialMediaButtonsProps) => {
     <>
       <button
         id="facebookButton"
-        onClick={() =>
-          showFacebookShare(encodedMarketUrl, encodedMarketDescription)
-        }
+        onClick={() => {
+          props.sendFacebookShare(props.marketAddress, props.marketDescription);
+          showFacebookShare(encodedMarketUrl, encodedMarketDescription);
```
Comment (bthaile) [inline packages/augur-ui/src/modules/market/components/common/social-media-buttons.tsx]: this looks weird can ` showFacebookShare(encodedMarketUrl, encodedMarketDescription);` be in the props.sendFacebookShare ... method

**Review by bthaile — APPROVED** [review]
super cool, lgtm

**Pull request merged**